### ISSUE TITLE
Unify document loading in Context Processing?

### ISSUE THREAD

**Issue by helins** [w3c/json-ld-api#584]
The Context Processing Algorithm deals with dereferencing remote documents at 2 occasions: `5.2` and `5.6`. Both look very similar to the point where I do not understand why there are seemingly differences:

- `5.2.1` details how resolving the `context` string against `base URL` might fail ; `5.6.3` does not mention anything about failure when resolving the value of `@import` against `base URL`
- `5.2.4` is very explicit about not dereferencing a URL that has already been dereferenced ; `5.6.4` goes straight into dereferencing without any further consideration

**Comment by gkellogg** [comment]
5.6.5 says what to do if loading the document fails; this would include cases where the URL is invalid.

5.2.4 is explicit about not dereferencing again, due to potential recursion loops. I don't think that the `@import` case had the same concern (@dlongley may recall), but implementations are generally encouraged to cache the result of retrieving external documents, which could be considered an expected optimization.

**Comment by helins** [comment]
Thanks @gkellogg. I guess not but would it make sense at all to also replicate `5.2.3` in `6`, WRT to `remote contexts`?

**Comment by pchampin** [comment]
> 5.6.5 says what to do if loading the document fails; this would include cases where the URL is invalid.

True, but that's a different error. 5.2.1 tells you what to do if the base URL is invalid, and therefore *resolving* the context URL against it is not possible. 5.6.3 does not tell you what to do in this situation... @helins has a point, we should probably render both steps more homogeneous.

> implementations are generally encouraged to cache the result of retrieving external documents, which could be considered an expected optimization.

I agree, but again, the fact that it is explicitly mentioned in one part, and not in another, may confuse implementers (as demonstrated by this issue).

**Comment by gkellogg** [comment]
<details>
<summary>Discussed on [today's call](https://json-ld.org/minutes/2024-07-10/#52)</summary>
<div onmouseout="$('#link-52').hide()" onmouseover="$('#link-52').show()" id="52" class="information"><a href="https://github.com/w3c/json-ld-api/issues/584">https://github.com/w3c/json-ld-api/issues/584</a> -&gt; Issue 584 Unify document loading in Context Processing? (by helins) [spec:editorial] [ErratumRaised] <a id="link-52" style="display:none;" href="#52">✪</a></div>
<div onmouseout="$('#link-53').hide()" onmouseover="$('#link-53').show()" id="53" class="comment"><span class="name">Gregg Kellogg</span>:  Perhaps a chance to DRY up the text. <a id="link-53" style="display:none;" href="#53">✪</a></div>
<div onmouseout="$('#link-54').hide()" onmouseover="$('#link-54').show()" id="54" class="comment"><span class="name">Pierre-Antoine Champin</span>:  I&apos;m hesitant to extract the common functionality, so maybe just go for consistency so as not to disrupt the flow of the document. <a id="link-54" style="display:none;" href="#54">✪</a></div>
</summary>
